src/mito/src/table.rs

```diff
@@ -332,6 +333,33 @@ impl<R: Region> Table for MitoTable<R> {
         Ok(())
     }
 
+    async fn compact(
+        &self,
+        region_number: Option<RegionNumber>,
+        wait: Option<bool>,
+    ) -> TableResult<()> {
+        let compact_ctx = wait.map(|wait| CompactContext { wait }).unwrap_or_default();
+        let regions = self.regions.load();
+
+        if let Some(region_number) = region_number {
+            if let Some(region) = regions.get(&region_number) {
+                region
+                    .compact(&compact_ctx)
+                    .await
+                    .map_err(BoxedError::new)
+                    .context(table_error::TableOperationSnafu)?;
+            }
+        } else {
+            let _ = futures::future::try_join_all(
```

Comment:
```suggestion
            futures::future::try_join_all(
```
no need for discard now 🥹 the old world has returned